src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -128,7 +128,14 @@ public async Task<string> DetermineNodeRuntimeVersion(IExecutionContext executio
 
             var output = new List<string>();
             var execExitCode = await dockerManager.DockerExec(executionContext, Container.ContainerId, string.Empty, osReleaseIdCmd, output);
-            string nodeExternal;
+            
+            string nodeExternal = preferredVersion;
+            if (executionContext.Global.EnvironmentVariables != null)
```

Comment:
We should get variables from the runner machine (host context), here, not the[ job level envs](https://docs.github.com/en/actions/learn-github-actions/environment-variables)

Reply:
I think we can do something like

```
bool.TryParse(Environment.GetEnvironmentVariable(Constants.Variables.Agent.ForcedActionsNodeVersion), out forceNode16);
```

Reply:
To test this, what you can do is create an `.env` file at the root of the runner layout folder, and set this value, then validate you see it works as expected